src/librustc_interface/interface.rs

```diff
@@ -24,6 +24,39 @@ use std::sync::{Arc, Mutex};
 
 pub type Result<T> = result::Result<T, ErrorReported>;
 
+pub enum RegisterLints {
+    Register(Box<dyn Fn(&Session, &mut LintStore) + Send + Sync>),
+    Empty,
```

Comment:
Instead of this being an enum it could also be a struct and we initialize it with `Box::new(move(|_, _| {}))`. I don't think this match is going to give us any measuable perf improvements over creating an empty closure.

Reply:
Hm yea, this is such a small use case it's probably not worth it to create another enum; I'll just close this for now